engine/src/widget-events.cpp

```diff
@@ -496,26 +508,121 @@ void MCWidgetEventManager::event_paint(MCWidget *p_widget, MCGContextRef p_gcont
     MCWidgetOnPaint(p_widget -> getwidget(), p_gcontext);
 }
 
-void MCWidgetEventManager::event_touch(MCWidget* p_widget, uint32_t p_id, MCEventTouchPhase p_phase, int2 p_x, int2 p_y)
+Bool MCWidgetEventManager::event_touch(MCWidget* p_widget, uint32_t p_id, MCEventTouchPhase p_phase, int2 p_x, int2 p_y)
 {
+    if (!widgetIsInRunMode(p_widget))
+    {
+        return False;
+    }
+    
+    /* If a touch is just starting, and this is the first touch then we need
+     * to compute the touched widget. */
+    MCWidgetRef t_touched_widget = m_touched_widget;
+    if (p_phase == kMCEventTouchPhaseBegan)
+    {
+        if (t_touched_widget == nullptr)
+        {
+            t_touched_widget = hitTest(p_widget->getwidget(), p_x, p_y);
+        }
+    }
+    
+    /* If we don't have a touched widget, then ignore the touch. */
+    if (t_touched_widget == nullptr)
+    {
+        return False;
```

Comment:
@runrevmark so in this case I think the widget will just receive the messages in LCS. Should the target be changed to the underlying layer?

Reply:
The caller of event_touch has special code in it to handle widgets. If a widget is the target of the first touch, that touch-sequence is then bound to that widget (at the event dispatch step). This branch will only happen if it is not a begin phase, and 'cancel_touches' has been called in the event manager - in which case you don't want to take any further action.

Reply:
What I mean is because the `return` is only if `event_touch` is `true` [here](https://github.com/livecode/livecode/pull/6313/files#diff-7d875e990369273324bbf33f95262ce7R1297) then it will continue and send the LCS messages. Perhaps it should return there regardless of the return value of `event_touch`?